modules/gke-cluster-standard/main.tf

```diff
@@ -419,13 +429,27 @@ resource "google_gke_backup_backup_plan" "backup_plan" {
       }
     }
 
-    all_namespaces = lookup(each.value, "namespaces", null) != null ? null : true
+    all_namespaces = lookup(each.value, "namespaces", null) != null || lookup(each.value, "applications", null) != null ? null : true
     dynamic "selected_namespaces" {
       for_each = each.value.namespaces != null ? [""] : []
       content {
         namespaces = each.value.namespaces
       }
     }
+    dynamic "selected_applications" {
+      for_each = each.value.applications != null ? [""] : []
+      content {
+        dynamic "namespaced_names" {
+          for_each = local.bkp_namespaces_object_map[each.key]
+          iterator = ns_obj_couple
+          content {
+            namespace = ns_obj_couple.value["namespace"]
+            name      = ns_obj_couple.value["object_name"]
+          }
+        }
+      }
```

Comment:
you're making this more complicated than necessary with the locals. The following should be enough

```hcl
        dynamic "namespaced_names" {
          for_each = flatten([for k, vs in each.value.applications : [
            for v in vs : { namespace = k, name = v }
          ]])
          content {
            namespace = namespaced_names.value.namespace
            name      = namespaced_names.value.name
          }
        }
```

Reply:
You're right, just pushed your suggest. I thought we need to avoid too many logic in resource block. Thank you (:

Reply:
Usually, yes. But this one was simple enough